k9mail/src/main/java/com/fsck/k9/helper/MailToHelper.java

```diff
@@ -0,0 +1,157 @@
+package com.fsck.k9.helper;
+
+import android.net.MailTo;
+import android.net.Uri;
+import android.util.Log;
+
+import com.fsck.k9.K9;
+import com.fsck.k9.exceptions.ParseException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * MailTo helper class
+ */
+public class MailToHelper {
+
+    static public final String MAILTO_SCHEME = "mailto";
+
+    // All the parsed content is added to the headers.
+    private HashMap<String, String> mHeaders;
+
+    // Well known headers
+    static private final String TO = "to";
+    static private final String BODY = "body";
+    static private final String CC = "cc";
+    static private final String BCC = "bcc";
+    static private final String SUBJECT = "subject";
+
+    public static boolean isMailTo(Uri uri) {
+        if(uri == null) {
+            return false;
+        }
+
+        return MAILTO_SCHEME.equals(uri.getScheme());
+    }
+
+    public static MailToHelper parse(Uri uri) throws ParseException {
+        if(uri == null || uri.toString() == null){
+            throw new NullPointerException();
```

Comment:
If you explicitly throw a `NullPointerException` please add a message argument to make it easy to distinguish from exceptions generated by the runtime.